defopt.py

```diff
@@ -131,6 +131,9 @@ def _create_parser(funcs, **kwargs):
     else:
         subparsers = parser.add_subparsers()
         for func in funcs:
+            if isinstance(funcs, dict):
+                name, func = func, funcs[func]
+                func.__name__ = name.replace("_", "-")
```

Comment:
Please don't modify the `__name__` of the function as this breaks aliasing for example, plus it kind of goes against defopt's design of not touching the callables.

Reply:
I agree on both counts. Unfortunately this doesn't seem to be as simple as deleting the line: `func.__name__ = name.replace("_", "-")`, as doing so causes most of the tests to fail. I'll have to look more carefully at the rest of the code to figure out the right way of doing this.